app/services/course/assessment/achievement_conditional_service.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# This service preloads all Achievement conditionals which lists Assessments as conditions.
+# Used for Assessments#Index to reduce n+1 queries.
+class Course::Assessment::AchievementConditionalService
```

Comment:
How about call this `AchievementPreload` service ? I think current name is too general and does not match is purpose.

I remember we have a service to preload tutors, maybe it's good to call it `TutorPreload` service too, @kxmbrian What you think ?